source/hardware/controllers/native_instruments_traktor_kontrol_s3.rst

```diff
@@ -0,0 +1,173 @@
+
+.. versionadded:: 2.3.0
+
+Compatibility
+-------------
+
+Controller
+~~~~~~~~~~
+
+The Kontrol S3 is a USB class compliant audio and :term:`HID` device,
+so it is compatible with Mixxx without any proprietary drivers on
+GNU/Linux and Mac OS X. On Windows, it is recommended to install the
+`driver from Native
+Instruments <https://www.native-instruments.com/en/support/downloads/drivers-other-files/#traktorkontrols3>`__
+and select the ASIO :ref:`sound API <preferences-sound-api>` in the :ref:`Sound Hardware section of Mixxx's Preferences <preferences-sound-hardware>`.
+
+With the S3 plugged in, the device is listed as an available
+controller in Mixxx's Preferences. The controller uses
+HID for the knobs, buttons, and other components on the device, so the
+mapping can only be loaded when you select the HID device on the left
+side of Mixxx's Preferences.
+
+Audio Hardware setup
+--------------------
+
+The S3 has a standard 4 channel sound card.  You should configure the channels
+as follows:
+
+============  =====================
+Audio Output  Channel Configuration
+============  =====================
+Master        Channel 1-2
+Headphones    Channel 3-4
+============  =====================
+
+Mapping description
+-------------------
+
+Note that Mixxx doesn't have the concept of a single "leader" deck for
+sync. Instead, push and hold the :hwlabel:`SYNC` button to "lock" sync on for all
+decks you want to remain in sync. Or you can push :hwlabel:`SHIFT` + :hwlabel:`SYNC` to lock
+sync on. Refer to :ref:`Sync Lock documentation <master-sync>`
+for details.
+
+.. hint:: There are two :hwlabel:`SHIFT` buttons, one for each side of the
+          controller. Make sure you press the :hwlabel:`SHIFT` button on the
+          same side as the control you want to modify.
+
+Mixer
+~~~~~
+
+  - The :hwlabel:`GAIN` and equalizer :hwlabel:`HIGH`/:hwlabel:`MID`/:hwlabel:`LOW` knobs and the :hwlabel:`CUE` (headphones) button behave as labelled.
+  - :hwlabel:`FX Enable` buttons: See Effect section below.
+  - The :hwlabel:`FILTER` knob controls the Quick Effect superknob. By default, this uses a high-/low-pass filter, but a different effect can be chosen in the :ref:`Equalizer section of Mixxx' Preferences <preferences-equalizers>`.
+  - :hwlabel:`EXT`: The :hwlabel:`EXT` button changes the fourth channel pregain (knob), pfl, and volume (slider) adjustments to operate with the Microphone input.  The microphone does not respond to EQ or effects.
+  - :hwlabel:`SHIFT` + :hwlabel:`EXT`: Switches input from Mic to Line and back again.
+
+The Main Volume knob on the S3 controls the volume of the S3's main
+output in hardware, so it does not affect the software main output gain knob
+in Mixxx by default. You can hold :hwlabel:`SHIFT` and turn the knob to adjust Mixxx'
+main output gain.  Note that this will still also adjust the hardware gain, so
+after you release :hwlabel:`SHIFT` you'll need to adjust the knob again.
```

Comment:
I wouldn't have bothered mapping this, but I suppose it doesn't hurt?

Reply:
yeah it should be fine.